frontend/src/components/helpers/auth/PersistentLogin.js

```diff
@@ -7,14 +7,31 @@ import { SocketMessages } from "../socket-messages/SocketMessages";
 import { GenericLoader } from "../../generic-loader/GenericLoader";
 import { PromptRun } from "../../custom-tools/prompt-card/PromptRun";
 
+let selectedProductStore;
+let selectedProduct;
+try {
+  selectedProductStore = require("../../../plugins/llm-whisperer/store/select-product-store.js");
+} catch {
+  // Ignore if hook not available
+}
+
 function PersistentLogin() {
   const [isLoading, setIsLoading] = useState(true);
   const { sessionDetails } = useSessionStore();
   const checkSessionValidity = useSessionValid();
 
+  try {
+    if (selectedProductStore?.useSelectedProductStore) {
+      selectedProduct = selectedProductStore?.useSelectedProductStore(
+        (state) => state?.selectedProduct
+      );
+    }
+  } catch (error) {
+    // Do nothing
+  }
   useEffect(() => {
     let isMounted = true;
-
+    setIsLoading(true);
```

Comment:
@jagadeeswaran-zipstack Why are we setting the `isLoading` state to `true` here? Can you provide some context?

Reply:
When page loads for first time loader will be set to false. since its a wrapper component its state remains same. When we navigate to other platform the loader state will be false which results to unexpected behaviours.

Reply:
@jagadeeswaran-zipstack Thanks for the explanation! I have a suggestion regarding the `isLoading` state. Instead of setting `isLoading` to `true` every time the `useEffect` is triggered, we should set it to `true` only when we call the `verifySession` function. This will prevent unnecessary state updates and re-renders when the session is already valid. Please review the changes and update the code accordingly. Let me know if any further clarification is needed.

```
useEffect(() => {
  let isMounted = true;

  const verifySession = async () => {
    try {
      await checkSessionValidity();
    } finally {
      isMounted && setIsLoading(false);
    }
  };

  if (!sessionDetails?.isLoggedIn) {
    setIsLoading(true); // Only trigger loading if session is invalid
    verifySession();
  } else {
    setIsLoading(false);
  }

  return () => (isMounted = false);
}, [selectedProduct]);
```

Reply:
@tahierhussain Yes understood. This approach is clean and clear. Thanks